### PR TITLE
parameter for square and cubic waveshaper

### PR DESCRIPTION
**Pull request by friedolino78**

### CODE REVIEW

**Comment by friedolino78** [comment]
complement to https://github.com/mruby-zest/mruby-zest/pull/38

File: src/Misc/WaveShapeSmps.cpp [hunk 1]
```diff
@@ -13,6 +13,7 @@
 
 #include "WaveShapeSmps.h"
 #include <cmath>
+#define X smps[i]
```
Comment (fundamental) [inline src/Misc/WaveShapeSmps.cpp:16]: Yeah, I'm going to recommend against this. If you have loops where the expression is used a number of times feel free to have "float &X = smps[i];", but listing this as a macro across the whole file makes it easy to mistakenly use and makes some of the code marginally less obvious.

That said, I do like that it makes some the further downwind changes easier to read, so as long as there's the "float &X = smps[i]" at the right scope, the smps[i]->X changes seem to be a net gain in readability.

**Comment by fundamental** [comment]
After the macro is changed to a set of properly scoped references, this should be good to merge.

**Comment by friedolino78** [comment]
the patches fe1754f and ff47100 has gone obsolete due to a less cpu intensive solution in [https://github.com/zynaddsubfx/zynaddsubfx/pull/62](https://github.com/zynaddsubfx/zynaddsubfx/pull/62)

**Comment by friedolino78** [comment]
i guess for the float &X = smps[i] thing it's easier to make it new from current master.

**Comment by fundamental** [comment]
I recall that other PR, but did that also add the different defaults as well? If so, then I guess this should be closed. If not then I guess it needs to be reworked?

**Comment by friedolino78** [comment]
the different defaults are only needed for the proposed linearization parameters for square and cubic. In PR62 I solved the linearization more elegant and less cpu costly by simply allowing drive values < 1.0
So no parameters -> no default values.

**Comment by fundamental** [comment]
Ah, ok. Just to make sure I'm not misinterpreting anything. Everything here is out-of-date and can be safely closed, right?

**Comment by friedolino78** [comment]
yes. only missing thing would be reminding me to "&X = smps[i]" ;-)

**Pull request closed without merging**